### PR TITLE
Fix/handle editor play mode state change

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# Reality Collective - Reality Toolkit Pull Request

## Overview

Fixes a VERY annoying issue whereby on entering Play Mode, Unity will run "Validate" and attempt to initialise the Service Manager. Where this is normally required for normal Unity Editing, this instance is then destroyed before "Awake" is called.
This fix "finally" allows the Service Manager to "skip" initialisation if the Unity Editor enters play mode and prevents double initialization.

Does not affect normal "Play" mode as this is an editor only behaviour.

Also fixed a minor grievance where the Service Manager for whatever reason is not initialised in the Editor and causes the inspector to throw a null reference exception (rare issue)

## Changes

- Implemented a "EditorPlayModeStateChange" editor flag, which is set on entering play mode in the editor and reset when it leaves.
- Fixed an edge case where Unity does NOT initialise the Service Framework in the editor that caused a null reference exception in the inspector.

## Breaking Changes

None

### CODE REVIEW

File: Editor/ServiceProfileInspector.cs [hunk 1]
```diff
@@ -375,7 +375,7 @@ private void DrawConfigurationOptionElement(Rect rect, int index, bool isActive,
                 serializedObject.ApplyModifiedProperties();
 
                 //TODO
```
Comment (FejZa) [inline Editor/ServiceProfileInspector.cs:377]: Can we get rid of this TODO comment?

File: Runtime/Utilities/ServiceFrameworkStatus.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Reality Collective. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace RealityCollective.ServiceFramework
```
Comment (FejZa) [inline Runtime/Utilities/ServiceFrameworkStatus.cs:4]: This sounds like it should live in an Editor assembly

File: Runtime/Utilities/ServiceManagerInstance.cs [hunk 3]
```diff
@@ -62,7 +65,8 @@ internal void InitialiseServiceManager()
 #if UNITY_EDITOR
         private void OnValidate()
         {
-            if(EditorApplication.isPlaying || !gameObject.activeInHierarchy || !enabled)
+            var UnityEditorRunState = EditorPrefs.GetBool(ServiceFrameworkStatus.UnityEditorRunStateKey);
```
Comment (FejZa) [inline Runtime/Utilities/ServiceManagerInstance.cs:68]: this needs to be camelCase

**Review by FejZa — CHANGES_REQUESTED** [review]

File: Runtime/Utilities/ServiceFrameworkStatus.cs (hunk 2 above)
Comment (SimonDarksideJ) [reply Runtime/Utilities/ServiceFrameworkStatus.cs:4]: Can't because it is needed in the runtime project :S  Unfortunately

**Review by FejZa — DISMISSED** [review]

**Comment by SimonDarksideJ** [comment]
Unbelievable, I think I found another edge case in the editor where it attempts to run both Validate and Awake at the same time.  Still need to test more, but on opening a scene when the editor loads.
Doesn't seem to happen if you just open a scene, just if your previous scene you had open when Unity closed, if unity actually reopens on the same scene, this issue can occur.
